entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityTypeHandler.kt

```diff
@@ -12,14 +12,37 @@ import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
+import java.util.Optional
 
 @RestController
 @RequestMapping("/ngsi-ld/v1/types")
 class EntityTypeHandler(
     private val entityTypeService: EntityTypeService
 ) {
 
+    /**
+     * Implements 6.25 - Retrieve Available Entity Types
+     */
+    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun getTypes(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @RequestParam details: Optional<Boolean>
+    ): ResponseEntity<*> {
+        val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
+        val mediaType = getApplicableMediaType(httpHeaders)
+        val detailedRepresentation = details.orElse(false)
+
+        val availableEntityTypes: Any = if (detailedRepresentation)
+            entityTypeService.getEntityTypes(listOf(contextLink))
+        else
+            entityTypeService.getEntityTypeList(listOf(contextLink))
+
+        return buildGetSuccessResponse(mediaType, contextLink)
```

Comment:
maybe the other endpoint needs to be fixed in the same way? if so, deal with it in another PR.

Reply:
Yes, okay.